hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -2364,6 +2364,14 @@ public UpdateFavoredNodesResponse updateFavoredNodes(RpcController controller,
   public BulkLoadHFileResponse bulkLoadHFile(final RpcController controller,
       final BulkLoadHFileRequest request) throws ServiceException {
     long start = EnvironmentEdgeManager.currentTime();
+    if(request.getClusterId() != null &&
+        request.getClusterId().equals(this.regionServer.clusterId)){
+      return BulkLoadHFileResponse.newBuilder().setLoaded(true).build();
+    }
+    String clusterId = request.getClusterId();
```

Comment:
nit. Do this a few lines higher and then you can replace a few request.getClusterId() calls.

Reply:
Yep, have addressed it on the last changes.